src/Confluent.Kafka/Serialization/FloatDeserializer.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright 2016-2017 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+
+namespace Confluent.Kafka.Serialization
+{
+    /// <summary>
+    ///     A deserializer for big endian encoded (network byte ordered) System.Single values.
+    /// </summary>
+    public class FloatDeserializer : IDeserializer<float>
+    {
+        /// <summary>
+        ///     Deserializes a big endian encoded (network byte ordered) System.Single value from a byte array.
+        /// </summary>
+        /// <param name="data">
+        ///     A byte array containing the serialized System.Single value (big endian encoding)
+        /// </param>
+        /// <returns>
+        ///     The deserialized System.Single value.
+        /// </returns>
+        public float Deserialize(byte[] data)
+        {
+            if (data == null)
+            {
+                throw new ArgumentException($"Arg [{nameof(data)}] is null");
+            }
+
+            if (data.Length != 4)
+            {
+                throw new ArgumentException($"Size of {nameof(data)} received by {nameof(FloatDeserializer)} is not 4");
+            }
+
+            float result = default(float);
+            // network byte order -> big endian -> most significant byte in the smallest address.
+            unsafe
+            {
```

Comment:
Note that corefx just changed their implementation of BitConverter (don't know when it will be released though), using directly IL (https://github.com/dotnet/corefx/blob/master/src/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.il and https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/BitConverter.cs#L126)

I wonder if it's better to use BitConverter rather than doing it ourself, especially now that we are cross patform (nothing in C# spec say that int/float it are little endian) The same goes for the other serialization class (we have time to see it coming though, and I previously benchmarked in an other project it was way quicker to use your way)